server.go

```diff
@@ -167,16 +167,27 @@ func (server *mongoServer) Connect(timeout time.Duration) (*mongoSocket, error)
 	case !dial.isSet():
 		// Cannot do this because it lacks timeout support. :-(
 		//conn, err = net.DialTCP("tcp", nil, server.tcpaddr)
-		conn, err = net.DialTimeout("tcp", server.ResolvedAddr, timeout)
+		if _, isunix := server.raddr.(*net.UnixAddr); isunix {
```

Comment:
Do you need to cast here?
radr. Network() should give you the network string ("unix", tcp") so it could be done in one line of code, no ?